File: Server/EndPoints/Hubs/client-dist/mytesthub.js
(function (signalR,$) {
signalR.createMyTestHub = function (loglevel) {
if (!loglevel) {
loglevel = signalR.LogLevel.Information;
}
var hub = new signalR.HubConnectionBuilder()
.withUrl('/hub/MyTestHub'.toLowerCase())
.configureLogging(loglevel)
    .build(); 
    
hub.server = {};
hub.client = {};

hub.server['onConnectedAsync'] = function() {return hub.invoke.apply(hub, $.merge(['OnConnectedAsync'], $.makeArray(arguments)));};
hub.server['onDisconnectedAsync'] = function(exception) {return hub.invoke.apply(hub, $.merge(['OnDisconnectedAsync'], $.makeArray(arguments)));};
hub.server['sendMessage'] = function(user,message) {return hub.invoke.apply(hub, $.merge(['SendMessage'], $.makeArray(arguments)));};
hub.server['test'] = function(index) {return hub.invoke.apply(hub, $.merge(['Test'], $.makeArray(arguments)));};
hub.server['testService'] = function(clientMessage) {return hub.invoke.apply(hub, $.merge(['TestService'], $.makeArray(arguments)));};
return hub;
}})(signalR,jQuery);

File: Server/EndPoints/Hubs/MyTestHub/MyTestHub.cs
using System;
using System.Threading.Tasks;
using Server.Core;
using Server.EndPoints.Hubs.GameHub;

namespace Server.EndPoints.Hubs.MyTestHub
{
    //[HubName("MyTestHub")]
    public class MyTestHub : Microsoft.AspNetCore.SignalR.Hub
    {
        private readonly IServiceProvider _svp;
        private readonly ITestSrvice _myTestSrvice;
        public MyTestHub(IServiceProvider svp)
        {
            _svp = svp;
            _myTestSrvice = (ITestSrvice) _svp.GetService(typeof(ITestSrvice));
        }

        public override async Task OnConnectedAsync()
        {
            await Groups.AddToGroupAsync(Context.ConnectionId, "All");
            await base.OnConnectedAsync();
        }
        public override async Task OnDisconnectedAsync(Exception exception)
        {
            await Groups.RemoveFromGroupAsync(Context.ConnectionId, "All");
            await base.OnDisconnectedAsync(exception);
        }
        public async Task SendMessage(string user, string message)
        {
 
            await Clients.All.SendCoreAsync("ReceiveMessage",  new object[2]{ user, message});
        }

        public Task<string> Test(int index)
        {

            Clients.All.InvokeAsync("TestSended", index);
            return Task.Factory.StartNew(() => "Hi return");
        }

        public Task<string> TestService(string clientMessage)
        {

            var responce = new
            {
                clientMessage = clientMessage,
                serverMessage = _myTestSrvice.GetMessage("test data from servers"),
            };
  
            Clients.All.InvokeAsync("Test", responce);
  
            return Task.Factory.StartNew(() => "Hi return");

        }

    }
}
